streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -202,16 +232,21 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
   /** Actor to receive messages from the receivers. */
   private class ReceiverTrackerActor extends Actor {
     def receive = {
-      case RegisterReceiver(streamId, typ, host, receiverActor) =>
-        registerReceiver(streamId, typ, host, receiverActor, sender)
-        sender ! true
-      case AddBlock(receivedBlockInfo) =>
-        sender ! addBlock(receivedBlockInfo)
-      case ReportError(streamId, message, error) =>
-        reportError(streamId, message, error)
-      case DeregisterReceiver(streamId, message, error) =>
-        deregisterReceiver(streamId, message, error)
-        sender ! true
+      // Actor stops to receive when tracker is stopping
+      if (!isTrackerStopping) {
```

Comment:
I dont get the logic here. If the tracker is in "stopping", then it should receive deregister messages. But stops it from receiving, isnt it?